aten/src/ATen/native/native_functions.yaml

```diff
@@ -5669,6 +5669,20 @@
     CPU: legacy::cpu::_th_histc
     CUDA: _histc_cuda
 
+- func: histogram.uniformbins(Tensor self, int bins=10, Tensor? weights=None, Scalar min=0, Scalar max=0, bool density=False) -> Tensor
+  use_c10_dispatcher: full
+  variants: method, function
+  dispatch:
+    CPU: _histogram_cpu
```

Comment:
Before digging too deep, can you elaborate on why this needs custom CPU and CUDA implementations, as opposed to a device generic implementation that applies to both CPU and CUDA?

Reply:
I did need a custom implementation for histogram with uniform bins on the CPU, as I could only use bincount there, while I could get a CUDA implementation for free using the CUDA_tensor_histogram template. I did try using a device generic implementation, but it has resulted in poor performance.

As for the version with custom bins, noticed that with the current code, there is no need for both, as the only difference between them is that the CUDA implementation gives the atomic_add warning. Will fix that in the next commit.

Reply:
Interesting. For small divergences you can also just use an if statement in the generic code (`if (t.device_type() == 'kCPU') {...`).